sway/input/seatop_down.c

```diff
@@ -99,28 +186,38 @@ static const struct sway_seatop_impl seatop_impl = {
 	.pointer_axis = handle_pointer_axis,
 	.tablet_tool_tip = handle_tablet_tool_tip,
 	.tablet_tool_motion = handle_tablet_tool_motion,
+	.touch_motion = handle_touch_motion,
+	.touch_up = handle_touch_up,
+	.touch_down = handle_touch_down,
 	.unref = handle_unref,
 	.end = handle_end,
 	.allow_set_cursor = true,
 };
 
 void seatop_begin_down(struct sway_seat *seat, struct sway_container *con,
-		uint32_t time_msec, double sx, double sy) {
-	seatop_begin_down_on_surface(seat, con->view->surface, time_msec, sx, sy);
+		double sx, double sy) {
+	seatop_begin_down_on_surface(seat, con->view->surface, sx, sy);
 	struct seatop_down_event *e = seat->seatop_data;
 	e->con = con;
 
 	container_raise_floating(con);
 	transaction_commit_dirty();
 }
 
+void seatop_begin_touch_down(struct sway_seat *seat,
+		struct wlr_surface *surface, struct wlr_touch_down_event *event,
+		double sx, double sy, double lx, double ly) {
+	seatop_begin_down_on_surface(seat, surface, sx, sy);
+	handle_touch_down(seat, event, lx, ly);
+}
+
 void seatop_begin_down_on_surface(struct sway_seat *seat,
-		struct wlr_surface *surface, uint32_t time_msec, double sx, double sy) {
+		struct wlr_surface *surface, double sx, double sy) {
 	seatop_end(seat);
 
 	struct seatop_down_event *e =
 		calloc(1, sizeof(struct seatop_down_event));
-	if (!e) {
+	if (!sway_assert(e, "Unable to allocate point_event")) {
```

Comment:
You copy pasted the log from the other `sway_assert`

Reply:
right, I didn't noticed the var name was used. Fixed

Reply:
I don't even know why this patch touch those lines anyway. I guess multiple rebase caused this. I should probably restore this simple return.